### ISSUE TITLE
Standard Notes crashing after upgrade

### ISSUE THREAD

**Issue by tcc56** [standardnotes/mobile#582]
I've been using Standard Notes on my phone for a while. It is installed from F-Droid. It works well, it syncs well on three devices.
After updating it (on Saturday I think), it crashes first thing when opening.
The App info says it's on version 3.14.0.
The phone uses Android v 10.

**Comment by AOSPFanboy** [comment]
I get the same thing. App version 3.14.0, running on GrapheneOS, Pixel 3. App is unusable. Tried enabling network permissions, reboot phone; nothing works. Unable to downgrade too for whatever reason.

**Comment by stefan-yas** [comment]
The crash should be resolved with version 3.14.1, however the F-Droid version might be lagging behind. Feel free to [sideload the .apk from here](https://github.com/standardnotes/mobile/releases/tag/3.14.1).

**Comment by Greymane96** [comment]
April 5 and still no update on FDroid.
I can't seem to sideload the .apk either. Do I need to uninstall first? I'm reluctant to do that because I don't want any chance of deleting any of my noted. 
Any advice is appreciated.

**Comment by moughxyz** [comment]
Are you using the app without an account? If with an account you can sign in after deletion. If without, I'm honestly not sure how different contexts are handled with regards to uninstalling. In my experience sometimes the data is deleted and sometimes it's not, but maybe someone with more Android-specific experience can corroborate or test out the behavior of deleting the app from Fdroid.

**Comment by Yannig123** [comment]
If you uninstall the app, all the notes will be gone. If you use it without an account and want to retrieve the data:
backup the file /data/data/com.standardnotes/databases/RKStorage
uninstall standardnotes, download an older apk, reinstall standardnotes, and move the file back to it's original location

**Comment by tcc56** [comment]
I downloaded app-prod-release.apk from https://github.com/standardnotes/mobile/releases/tag/3.14.1 to my phone using Firefox and clicked to install. App wasn't installed.
Went into settings > apps & notifications > advanced > select special app access > install unknown apps. Firefox (and F-Droid) were allowed to install unknown apps. This is puzzling, as I'm using Firefox to install the APK file, and it isn't working.
I tried enabling installing unknown apps from File Manager & Files. The 3.14.1 update to Standard Notes still didn't install.
I double checked. Standard Notes didn't install the 3.14.1 update & closes as soon as I open it.
This is as far as I can get on this.

**Comment by Greymane96** [comment]
Please help a noob.
I can not find the file " /data/data/com.standardnotes/databases/RKStorage " on my phone to create a backup. 
I have notes stored on my device only and will lose them all if I uninstall the FDroid version.  
Is there any hope for me? 
Any advice on locating this data file?
Should I wait for FDroid to publish another update and hope that the data will not be lost?
I feel like an idiot for storing so much important info on my device, never realizing that the data could be duplicated with an account.

Pixel 4a
Graphene OS

**Comment by dreamsyntax** [comment]
> Please help a noob. I can not find the file " /data/data/com.standardnotes/databases/RKStorage " on my phone to create a backup. I have notes stored on my device only and will lose them all if I uninstall the FDroid version. Is there any hope for me? Any advice on locating this data file? Should I wait for FDroid to publish another update and hope that the data will not be lost? I feel like an idiot for storing so much important info on my device, never realizing that the data could be duplicated with an account.
> 
> Pixel 4a Graphene OS

Assuming Android 12, scoped storage means you need root to access it

**Comment by Emajor69** [comment]
What's the ETA for the fdroid update?

**Comment by tcc56** [comment]
> @tcc56 Try backing up your data first and then uninstalling the F-Droid version. Then try to install the 3.14.1 version.

This worked like a charm.
I uninstalled the F-Droid version of Standard Notes on the smartphone and re-installed it by sideloading the .apk file. After signing in, it immediately synced with my Standard Notes installs on an ereader and laptop. I added a note to double check, and it synced.
Everything is back the way it should be.
Thank you for your help -- and thank you to all the contributors who make Standard Notes possible.

**Comment by stefan-yas** [comment]
> Any way you could either upload the F-Droid signature apk on the releases tab or update the F-Droid repo soon with 3.14.1? Its the main reason we are not able to upgrade without wiping due to the signature difference

@dreamsyntax As an organization, we're not maintaining the F-Droid packages on there. They should be updated to the newest version automatically after release, so we've reached out the package maintainer on F-Droid to see if we can get it up and running again.

**Comment by LAN-SoL0** [comment]
Same issue, can't find /data/data/com.standardnotes/databases/RKStorage on GrapheneOS Pixel 4a.  Unhid system folders, and even tried a Terminal emulator, can't find it at all.  Can't uninstall because I didn't use an account.  Glad this is getting attention, love you guys.

**Comment by Manodiestra** [comment]
> Same issue, can't find /data/data/com.standardnotes/databases/RKStorage on GrapheneOS Pixel 4a.  Unhid system folders, and even tried a Terminal emulator, can't find it at all.  Can't uninstall because I didn't use an account.  Glad this is getting attention, love you guys.

I use GrapheneOS, too. You can't access `/data/` on GrapheneOS. Nor will you ever without forking it and making your own version. Opening up access root is contrary to the design goals of GrapheneOS since it's primarily a security hardening project. Root access is only granted by API's designed with least privilege in mind.

If anyone else here is using GrapheneOS and Standard Notes offline from Fdroid and don't want to loose your notes, you're going to have to wait until 3.14.1 makes it to the Fdroid repos and hope that works.

I did the same thing because I kept having issues getting the android app to connect to my standard notes account.

Lesson learned, use Standard Notes onlime as intended, or make a backup somewhere else. Or both.

**Comment by dreamsyntax** [comment]
> @dreamsyntax As an organization, we're not maintaining the F-Droid packages on there. They should be updated to the newest version automatically after release, so we've reached out the package maintainer on F-Droid to see if we can get it up and running again.

Yikes. Will definitely not be using F-Droid for this app again and instead stick to the official Github releases.
Thanks again for all the effort trying to help us out.

**Comment by Greymane96** [comment]
Thanks for the blunt help.
Lesson learned.
Got an account and now everything is cool (except for all the lost
passwords, lists and notes)   :(
Thanks for your help, everyone.

**Comment by moughxyz** [comment]
Unfortunately this episode happens with F-Droid about once a year. They need to compile the app from source for every release, and sometimes with major React Native upgrades on our end, their CI chokes. I recommend bumping this thread to get someone to take a look: https://gitlab.com/fdroid/rfp/-/issues/379

**Comment by G1tbwoy** [comment]
> If you uninstall the app, all the notes will be gone. If you use it without an account and want to retrieve the data: backup the file /data/data/com.standardnotes/databases/RKStorage uninstall standardnotes, download an older apk, reinstall standardnotes, and move the file back to it's original location



> If you uninstall the app, all the notes will be gone. If you use it without an account and want to retrieve the data: backup the file /data/data/com.standardnotes/databases/RKStorage uninstall standardnotes, download an older apk, reinstall standardnotes, and move the file back to it's original location

I cannot access data/data folder, since I'm not root. So do I understand correctly that if I don't want to lose my notes all I can do is wait for an F-droid update?

**Comment by steaksweremade** [comment]
> A fix has been pushed but you will need to wait for a new build cycle.

How long does that usually take?

**Comment by omegis83** [comment]
So curious as to why it's failing in Build Status still on F-Droid

**Comment by proletarius101** [comment]
> Curious as to why it's failing in Build Status still on F-Droid
> 
> ![Screenshot_20220413-210412](https://user-images.githubusercontent.com/21134560/163299538-b6d5ab9a-69c0-4964-b2a0-67c656cd43eb.png)

Because it's days ago? A new build cycle hasn't started yet

**Comment by LAN-SoL0** [comment]
> I use GrapheneOS, too. You can't access `/data/` on GrapheneOS. Nor will you ever without forking it and making your own version. Opening up root access is contrary to the design goals of GrapheneOS since it's primarily a security hardening project. Root access is only granted by API's designed with least privilege in mind.
> 

Surprisingly enough, I can get to /data/data but no further.

**Comment by Manodiestra** [comment]
I have GrapheneOS and got the update on Fdroid and it seems to work just fine again. I think the issue can be closed.

**Comment by AOSPFanboy** [comment]
> I have GrapheneOS and got the update on Fdroid and it seems to work just fine again. I think the issue can be closed.

Same here - got the update on F-Droid this morning.

**Comment by dreamsyntax** [comment]
Agreed, just updated via F-Droid and all my data is still there. Thanks for the effort to get this resolved.

**Comment by realbiz21** [comment]
@dreamsyntax, abandoning F-Droid is the wrong move.

I also experienced this crashing once f-droid updated to 3.14 and knew the only thing to do was wait for a new build, as I used Standard Notes purely offline and on GrapheneOS, since the application data is impossible to retrieve.

I regularly export my notes but there were some I hadn't exported, so waiting was the only solution.

Yes, these situations are an inconvenience, but abandoning F-Droid takes us back to the days of downloading binaries without caring what's in them. Once we get too comfortable with that, there's no need for code to be "open source" anymore. Yes we can build the apps ourselves, but if we were doing that, we wouldn't be on this GitHub issue.
